omniscidb/QueryEngine/CostModel/ExtrapolationModels/LinearRegression.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+    Copyright (c) 2022 Intel Corporation
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+        http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include "LinearRegression.h"
+#include <armadillo>
+
+namespace costmodel {
+
+struct LinearRegression::PrivateImpl {
+  arma::mat buildFeaturesMatrix(const std::vector<Detail::Measurement>& measurement);
+  arma::vec buildTargets(const std::vector<Detail::Measurement>& measurement);
+
+  arma::vec w_;
```

Comment:
The name implies the member is private which is a bit confusing. I'd change it to something like `weights`.